Divine/CLI/CommandLineLogger.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using Divine.Enums;
+
+namespace Divine.CLI
+{
+    internal class CommandLineLogger
+    {
+        private static readonly LogLevel OptionLogLevel = CommandLineActions.LogLevel;
+
+        public static void LogFatal(string message, int errorCode) => Log(LogLevel.FATAL, message, errorCode);
+        public static void LogError(string message) => Log(LogLevel.ERROR, message);
+        public static void LogWarn(string message) => Log(LogLevel.WARN, message);
+        public static void LogInfo(string message) => Log(LogLevel.INFO, message);
+        public static void LogDebug(string message) => Log(LogLevel.DEBUG, message);
+        public static void LogTrace(string message) => Log(LogLevel.TRACE, message);
+        public static void LogAll(string message) => Log(LogLevel.ALL, message);
+
+        private static void Log(LogLevel loglevel, string message, int errorCode = -1)
+        {
+            if (OptionLogLevel == LogLevel.OFF)

```

Comment:
Hm, setting LogLevel to OFF will prevent this function from calling Environment.Exit(), so we'll always return a zero code.
... actually, we won't stop execution at all.

Reply:
I didn't catch that. Will fix.